schema/v1.1.0/dataset_config_models_extended.py

```diff
@@ -0,0 +1,321 @@
+from __future__ import annotations
+
+import asyncio
+import re
+from typing import Dict, List, Optional, Set, Tuple
+
+import aiohttp
+from async_lru import alru_cache
+from dataset_config_models import (
+    Annotation,
+    AnnotationConfidence,
+    AnnotationEntity,
+    AnnotationObject,
+    AnnotationSource,
+    Author,
+    Container,
+)
+from pydantic import field_validator, model_validator
+from typing_extensions import Self
+
+CELLULAR_COMPONENT_GO_ID = "GO:0005575"
+
+running_network_validation = False
+
+# ==============================================================================
+# Annotation Object Validation
+# ==============================================================================
+
+
+@alru_cache
+async def validate_go_id(go_id: str) -> Tuple[bool, dict]:
+    url = f"https://api.geneontology.org/api/ontology/term/{go_id}"
+    headers = {"Accept": "application/json"}
+    async with aiohttp.ClientSession() as session, session.get(url, headers=headers) as response:
+        go_id_data = await response.json()
+        return "label" in go_id_data or "description" in go_id_data, go_id_data
+
+
+@alru_cache
+async def is_go_id_ancestor(go_id_ancestor: str, go_id: str) -> bool:
+    url = f"https://api.geneontology.org/api/ontology/term/{go_id}/subgraph?start=0&rows=100"
+    headers = {"Accept": "application/json"}
+    async with aiohttp.ClientSession() as session, session.get(url, headers=headers) as response:
+        response_json = await response.json()
+        return go_id_ancestor in [ancestor["id"] for ancestor in response_json["ancestors"]]
+
+
+class ExtendedValidationAnnotationObject(AnnotationObject):
```

Comment:
classes are getting extended so that their existing validators built by linkml are preserved. note however that if fields are not copied over, weird behavior happens in some odd cases, so in the code following, the field values are re-assigned their value (in addition changing the field type to reference an extended-validation version of the original class)